### ISSUE TITLE
False positive of no-shadow rule with ENUMs

### ISSUE THREAD

**Issue by DuncanOgle** [typescript-eslint/typescript-eslint#2483]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "rules": {
    "no-shadow": ["error"]
  }
}
```

```TS
// This is the entire file - there is nothing else in it apart from these 4 lines
enum Direction {
    Up = 1,
    Down = -1
}
```

**Expected Result**

The ENUM should not state that it is declared in the upper scope

**Actual Result**

 The following error message is received;

`error  'Direction' is already declared in the upper scope  no-shadow`

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.0.1` |
| `TypeScript`                | `4.0.2` |
| `ESLint`                    | `7.8.1` |
| `node`                      | `12.16.3` |

Basically, it looks like the latest version of the parser fails to read ENUMs properly.

Downgrading `@typescript-eslint/parser` to `v3.10.1` with the following `.eslintrc` file works and does not report ENUMs being declared in the upper scope.

```json
{
  "parserOptions": {
    "sourceType": "module"
  },
  "parser": "@typescript-eslint/parser",
  "rules": {
    "no-shadow": ["error"]
  }
}
```

**Comment by glen-84** [comment]
You need to use `@typescript-eslint/no-shadow`.

**Comment by DuncanOgle** [comment]
Confirmed, disabling `no-shadow` and enabling `@typescript-eslint/no-shadow` works. Thanks

```
"rules": {
    "no-shadow": "off",
    "@typescript-eslint/no-shadow": "error"
}
```
